plasmapy/formulary/densities.py

```diff
@@ -0,0 +1,56 @@
+"""Functions to calculate plasma density parameters."""
+__all__ = [
+    "plasma_critical_density",
+]
+
+import astropy.units as u
+
+from astropy.constants.si import e, eps0, m_e
+
+from plasmapy.utils.decorators import validate_quantities
+
+
+@validate_quantities(
+    omega_p={"can_be_negative": False},
+    validations_on_return={
+        "units": [u.m**-3],
+        "equivalencies": [
+            (u.F * u.kg * u.rad**2 / (u.C**2 * u.m * u.s**2), u.m**-3)
+        ],
+    },
+)
+def plasma_critical_density(omega_p: u.rad / u.s) -> u.m**-3:
+    r"""Calculate the plasma critical density.
+
+    Parameters
+    ----------
+    omega_p : `~astropy.units.Quantity`
+        The plasma frequency in units of angular frequency.
+
+    Returns
+    -------
+    n_c : `~astropy.units.Quantity`
+        The plasma critical density.
+
+    Notes
+    -----
+    The plasma critical density is given by the formula
```

Comment:
How about adding that sentence of explanation you added to the test here to explain what this is physically?

"The plasma critical density is defined as the value at which the plasma frequency
        equals the frequency of radiation."

Although I would maybe revise slightly (in both places) 


"The plasma critical density _for a given frequency of radiation_ is defined as the value at which the _electron_ plasma frequency
        equals the frequency of _the_ radiation."